### PR TITLE
[APR] Order pools by the biggest TVL on pool list

### PR DESCRIPTION
**Pull request by luizakp**
- Add network paths
- Add `getPoolList` to get PoolsByTotalLiquidity

https://www.loom.com/share/1586f0f01e7d4e9484e68774f0ba5750?sid=b20c26b9-ca90-4f57-b0c3-7f789f65a9a5

### CODE REVIEW

File: apps/balancer-tools/src/app/apr/round/(components)/PoolListTable.tsx [hunk 1]
```diff
@@ -112,16 +137,26 @@ export function PoolListTable({ roundId }: { roundId: string }) {
...
+            tableData.map((gauge) => (
+              <TableRow
+                tableData={tableData}
+                setTableData={setTableData}
+                key={gauge.id}
+                poolId={gauge.id}
+                network={gauge.network}
+                roundId={roundId}
+              />
+            ))
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/apr/round/(components)/PoolListTable.tsx:158]: You can move the TableRow component to another file and put a Suspense boundary around it

File: apps/balancer-tools/src/app/apr/round/(components)/PoolListTable.tsx [hunk 2]
```diff
@@ -30,22 +29,48 @@ interface PoolStats {
...
+            };
+          }),
+        );
+
+        setTableData(poolListWithStats);
+
+        setIsLoadingFetchPoolList(false);
+      } catch (error) {
+        setIsLoadingFetchPoolList(false);
+      }
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/apr/round/(components)/PoolListTable.tsx:68]: ❌ this must be done on the server side

Comment (luizakp) [reply apps/balancer-tools/src/app/apr/round/(components)/PoolListTable.tsx]: I need some help with this. Tried to use server actions but without success :(

File: apps/balancer-tools/src/app/apr/[network]/round/(components)/PoolListTable.tsx [hunk 3]
```diff
@@ -30,22 +29,56 @@ interface PoolStats {
...
+  roundId,
+  network,
+}: {
+  roundId: string;
+  network: Network;
+}) {
+  const [isLoadingFetchPoolList, setIsLoadingFetchPoolList] = useState(true);
+  const [tableData, setTableData] = useState([] as PoolTableData[]);
   const [sortField, setSortField] = useState("");
   const [order, setOrder] = useState<"asc" | "desc">("asc");
```
Comment (Rawallon) [inline apps/balancer-tools/src/app/apr/[network]/round/(components)/PoolListTable.tsx:45]:
```suggestion
  const [sortField, setSortField] = useState("tvl");
  const [order, setOrder] = useState<"asc" | "desc">("desc");
```

File: packages/gql/src/balancer/pools.ts [hunk 4]
```diff
@@ -95,6 +95,24 @@ export const poolWhereBlockNumber = gql`
   }
 `;
 
+export const poolsByTotalLiquidity = gql`
+  query PoolsByTotalLiquidity($first: Int = 10, $poolIdList: [ID!]) {
+    pools(
+      where: { totalLiquidity_gt: 0, id_in: $poolIdList }
+      orderBy: totalLiquidity
+      orderDirection: desc
+      first: $first
```
Comment (Rawallon) [inline packages/gql/src/balancer/pools.ts:104]:
```suggestion
  query PoolsByTotalLiquidity($first: Int = 10, $skip: Int = 0, $poolIdList: [ID!]) {
    pools(
      where: { totalLiquidity_gt: 0, id_in: $poolIdList }
      orderBy: totalLiquidity
      orderDirection: desc
      skip: $skip
      first: $first
```

Comment (Rawallon) [reply packages/gql/src/balancer/pools.ts:104]: We could add skip here to add paging, right?

File: apps/balancer-tools/src/app/apr/page.tsx [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import { redirect } from "next/navigation";
 
 export default function Page() {
-  redirect("/apr/round/current");
+  redirect("/apr/ethereum/round/current");
```
Comment (ribeirojose) [inline apps/balancer-tools/src/app/apr/page.tsx:4]: why add the network name here?

**Review by ribeirojose — CHANGES_REQUESTED** [review]
blocking until the decision to add the chain param to the URL gets cleared out

**Comment by luizakp** [comment]
@ribeirojose @Rawallon handling this on #235

**Pull request closed without merging**